src/operators.jl

```diff
@@ -277,7 +277,8 @@ Base.sum(j::JuMPDict)  = sum(values(j.tupledict))
 Base.sum(j::JuMPArray{Variable}) = AffExpr(vec(j.innerArray), ones(length(j.innerArray)), 0.0)
 Base.sum(j::JuMPDict{Variable})  = AffExpr(collect(values(j.tupledict)), ones(length(j.tupledict)), 0.0)
 Base.sum(j::Array{Variable}) = AffExpr(vec(j), ones(length(j)), 0.0)
-function Base.sum{T<:GenericAffExpr}(affs::Array{T})
+Base.sum(j::AbstractArray{Variable}) = sum([j[i] for i in eachindex(j)]) # to handle non-one-indexed arrays.
```

Comment:
As long as ``vec(::AbstractArray{Variable})`` will reliably work, I think the definition [above](https://github.com/JuliaOpt/JuMP.jl/pull/980/files#diff-188d49b6a9fe32e3c270a948c830cbcfR279) should work fine.